### PR TITLE
add codecov.io service.

### PR DESCRIPTION
**Pull request by appleboy**
Add codecov.io coverage service. See https://codecov.io/gh/go-gitea/gitea

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -130,12 +130,13 @@ coverage: unit-test-coverage integration-test-coverage
...
+	echo "mode: set" > coverage.all
 	for PKG in $(PACKAGES); do\
-	  touch $$GOPATH/src/$$PKG/coverage.out;\
-	  egrep "$$PKG[^/]*\.go" integration.coverage.out > int.coverage.out;\
-	  gocovmerge $$GOPATH/src/$$PKG/coverage.out int.coverage.out > pkg.coverage.out;\
-	  mv pkg.coverage.out $$GOPATH/src/$$PKG/coverage.out;\
-	  rm int.coverage.out;\
+		egrep "$$PKG[^/]*\.go" integration.coverage.out > int.coverage.out;\
+		gocovmerge $$GOPATH/src/$$PKG/coverage.out int.coverage.out > pkg.coverage.out;\
+		grep -h -v "^mode:" pkg.coverage.out >>  coverage.all;\
```
Comment (appleboy) [inline Makefile:146]: Just add this line and convert space to tabs.

**Comment by tboerger** [comment]
Drone will get integrated coverage with 0.9

**Comment by lunny** [comment]
@tboerger we can convert back when drone 0.9 is ready.

**Comment by ethantkoenig** [comment]
FYI, `make coverage` is not run in CI as of #2492.

**Comment by appleboy** [comment]
@ethantkoenig Yes. it will be fixed in this commit: https://github.com/go-gitea/gitea/pull/2493/commits/18ef72f6be28e65bf454f93b17b829aba761df5c

**Comment by lafriks** [comment]
In generate_coverage there are lot of errors like `failed to parse profiles: open /srv/app/src/code.gitea.io/gitea/coverage.out: no such file or directory`

**Review by lafriks — APPROVED** [review]

**Comment by appleboy** [comment]
Only two conditions we need to generate the coverage report.

```
when:
   event: [ push ]
   branch: [ master ]
```

or 

```
when:
  event: pull_request
```

File: .drone.yml [hunk 2]
```diff
@@ -27,17 +27,42 @@ pipeline:
...
     group: test
     environment:
       TAGS: bindata sqlite
       GOPATH: /srv/app
     commands:
+      - make unit-test-coverage
+    when:
+      event: [ pull_request ]
+
+  test-master:
```
Comment (tboerger) [inline .drone.yml]: Why do you separate master and pull requests?

**Review by tboerger — CHANGES_REQUESTED** [review]

**Comment by tboerger** [comment]
You can combine the conditions when event: push, pull_request and branch: master are combined it triggers for pushs on master AND pull requests against master.

**Comment by tboerger** [comment]
Ah and beside that I would name the steps the same since anyway only one of them should be executed per build

**Comment by appleboy** [comment]
@tboerger Done

**Review by bkcsoft — APPROVED** [review]

**Comment by bkcsoft** [comment]
LGTM

**Review by tboerger — APPROVED** [review]

**Comment by tboerger** [comment]
LGTM

**Comment by lafriks** [comment]
@appleboy please rebase

**Comment by lafriks** [comment]
Your makefile needs additional changes after my PR was merged to generate mysql.ini from mysql.ini.tmpl

**Pull request merged**